### PR TITLE
docs: dockerize manual instrumentation example

### LINKED ISSUE #2798: [docs] containerize manual instrumentation example

### PR DESCRIPTION
**Pull request by axiomofjoy**
Dockerizes manual instrumentation example so someone can spin it up and try it out in a single command.

resolves #2798

### CODE REVIEW

File: examples/manually-instrumented-chatbot/chat-service/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+server:
+	uvicorn chat.app:app --reload
+
+format:
+	ruff check --fix . && ruff format .
+
+typecheck:
+	pyright .
```
Comment (mikeldking) [inline examples/manually-instrumented-chatbot/chat-service/Makefile:8]: Kinda curious why you chose pyright here.

File: examples/manually-instrumented-chatbot/chat-service/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	uv pip compile pyproject.toml > requirements.txt
+
+schema:
+	python export_schema.py
+
+docker-build:
+	docker build -t chat-service .
+
+docker-run:
+	docker run -e "OPENAI_API_KEY=$OPENAI_API_KEY" -p 8000:8000 -it chat-service
```
Comment (mikeldking) [inline examples/manually-instrumented-chatbot/chat-service/Makefile:20]: Doesn't the env variable get mounted anyways?

File: examples/manually-instrumented-chatbot/chat-service/export_schema.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from fastapi.openapi.utils import get_openapi
+
+from chat.app import app
+
+with open("../frontend/schema.json", "w") as f:
+    json.dump(
+        get_openapi(
+            title="Chat Service Schema",
+            version="1.0.0",
+            description="Your API Description",
```
Comment (mikeldking) [inline examples/manually-instrumented-chatbot/chat-service/export_schema.py:16]: Should fill this out?

**Review by mikeldking — APPROVED** [review]
Nice one!

File: examples/manually-instrumented-chatbot/chat-service/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.11-slim as builder
+WORKDIR /app
+COPY requirements.txt /app/
+RUN pip install uv
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply examples/manually-instrumented-chatbot/chat-service/Dockerfile:4]: `uv` is definitely faster, although the reason I chose it here is to use `uv pip sync`, which guarantees that we are installing the exact versions of the packages pinned in `requirements.txt`. `pip install` does not provide any guarantees here.

File: examples/manually-instrumented-chatbot/chat-service/Makefile (hunk 1 above)
Comment (axiomofjoy) [reply examples/manually-instrumented-chatbot/chat-service/Makefile:8]: Arbitrary choice. I can switch to `mypy`.

File: examples/manually-instrumented-chatbot/chat-service/Makefile (hunk 2 above)
Comment (axiomofjoy) [reply examples/manually-instrumented-chatbot/chat-service/Makefile:20]: As far as I know, I don't believe it does.

File: examples/manually-instrumented-chatbot/chat-service/export_schema.py (hunk 3 above)
Comment (axiomofjoy) [reply examples/manually-instrumented-chatbot/chat-service/export_schema.py]: addressed in 8752d4f8aed347787e7b6d510995c3427ac3bd0e

File: examples/manually-instrumented-chatbot/chat-service/Makefile (hunk 1 above)
Comment (axiomofjoy) [reply examples/manually-instrumented-chatbot/chat-service/Makefile:8]: I think `pyright` is faster, though.

**Pull request merged**